lib/db/schema/patch-020-021.sql

```diff
@@ -0,0 +1,35 @@
+CREATE PROCEDURE `sessionDevice_1` (
+  IN `inTokenId` BINARY(32)
+)
+BEGIN
+  SELECT
+    t.tokenData,
+    t.uid,
+    t.createdAt,
+    t.uaBrowser,
+    t.uaBrowserVersion,
+    t.uaOS,
+    t.uaOSVersion,
+    t.uaDeviceType,
+    t.lastAccessTime,
+    a.emailVerified,
+    a.email,
+    a.emailCode,
+    a.verifierSetAt,
+    a.locale,
+    a.createdAt AS accountCreatedAt,
+    d.id AS deviceId,
+    d.name AS deviceName,
+    d.type AS deviceType,
+    d.createdAt AS deviceCreatedAt,
+    d.callbackURL,
+    d.callbackPublicKey
+  FROM
+    sessionTokens t
+    LEFT JOIN accounts a ON t.uid = a.uid
+    LEFT JOIN devices d ON t.tokenId = d.sessionTokenId
+  WHERE
+    t.tokenId = inTokenId;
```

Comment:
The devices table has a unique index on (uid, sessionTokenId), and we should  do an `EXPLAIN` to confirm that MySQL is smart enough to use if for the `LEFT JOIN devices` here.  I wouldn't be surprised if we had to add a technically-redundant `d.uid = t.uid` in here somewhere, just to tell the query optimizer that it's safe to use this index.